king/king.wdl

```diff
@@ -0,0 +1,425 @@
+		docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File king_ibdseg_output = glob("*.seg")[0]
+	}
+
+}
+
+# [4] king_to_matrix -- Format initial kinship estimates from KING as a matrix
+
+# Add java script from cwl?
+
+task king_to_matrix {
+	input {
+		File king_file
+		File? sample_include_file
+
+		# optional
+		Float? sparse_threshold = 0.02209709
+		String? out_prefix
+		String? kinship_method
+		
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int king_size = ceil(size(king_file, "GB"))
+	Int final_disk_dize = king_size + addldisk
+
+	# Workaround for optional files
+	Boolean defSampleInclude = defined(sample_include_file)
+	
+	command {
+		set -eux -o pipefail
+
+		echo "Generating config file"
+		# noquotes = argument has no quotes in the config file to match CWL config EXACTLY
+		python << CODE
+		import os
+
+		f = open("king_to_matrix.config", "a")
+		f.write('king_file "~{king_file}"\n')
+
+		if "~{defSampleInclude}" == "true":
+			f.write('sample_include_file ""~{sample_include_file}""\n')
+
+		#f.write('sample_include_file "~{sample_include_file}"\n')
+
+		if "~{sparse_threshold}" != "":
+			f.write('sparse_threshold "~{sparse_threshold}"\n')
+		else:
+			f.write('sparse_threshold NA\n')
+
+		# check for empty string
+		if "~{out_prefix}" != "":
+			f.write('out_prefix "' + "~{out_prefix}" + '"\n')
+		else:
+			f.write('out_prefix "' + "~{basename(king_file, ".seg")}" + '_king_ibdseg_Matrix"\n')
+
+		if "~{kinship_method}" != "":
+			f.write('kinship_method "~{kinship_method}"\n')
+
+		if "~{kinship_method}" not in ['king_ibdseg', 'king_robust']:
+			f.write('kinship_method "king_ibdseg"\n')
+		else:
+			f.write('kinship_method "~{kinship_method}"\n')
+
+
+		f.close()
+		CODE
+
+		echo "Calling R script king_to_matrix.R"
+		Rscript /usr/local/analysis_pipeline/R/king_to_matrix.R king_to_matrix.config
+	}
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File king_matrix = glob("*.RData")[0]
+		File config_file = "king_to_matrix.config"
+	}
+
+}
+
+# [5] kinship_plots -- Make hexbin plots of the initial kinship estimates from KING
+
+task kinship_plots {
+	input {
+		File kinship_file
+		String? kinship_method
+
+		# optional
+		Float? kinship_plot_threshold = 0.04419417382
+		File? phenotype_file
+		String? group
+		File? sample_include_file
+		String? out_prefix = "kinship"
+		
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int kinship_size = ceil(size(kinship_file, "GB"))
```

Comment:
I think your outputs are always going to be relatively small, but still consider accounting for your optional inputs. phenotype_file in particular can be quite large.